src/java.base/share/classes/jdk/internal/classfile/impl/TerminalCodeBuilder.java

```diff
@@ -25,8 +25,24 @@
 package jdk.internal.classfile.impl;
 
 import java.lang.classfile.CodeBuilder;
+import java.lang.classfile.CodeModel;
+import java.lang.classfile.attribute.CodeAttribute;
 
 public sealed interface TerminalCodeBuilder extends CodeBuilder, LabelContext
         permits DirectCodeBuilder, BufferedCodeBuilder {
     int curTopLocal();
+
+    static int setupTopLocal(MethodInfo methodInfo, CodeModel original) {
+        int paramSlots = Util.maxLocals(methodInfo.methodFlags(), methodInfo.methodTypeSymbol());
+        if (original == null) {
+            return paramSlots;
+        }
+        if (original instanceof CodeAttribute attr) {
+            return Math.max(paramSlots, attr.codeLength());
```

Comment:
```suggestion
            return Math.max(paramSlots, attr.maxLocals());
```

Reply:
Good point, will add a test too to verify.